sidecar_acceptor_test.go

```diff
@@ -32,67 +31,46 @@ var (
 	}
 )
 
-func newTestDB(t *testing.T) (*clientdb.DB, func()) {
-	tempDir, err := ioutil.TempDir("", "client-db")
-	require.NoError(t, err)
-
-	db, err := clientdb.New(tempDir, clientdb.DBFilename)
-	if err != nil {
-		require.NoError(t, os.RemoveAll(tempDir))
-		t.Fatalf("unable to create new db: %v", err)
-	}
-
-	return db, func() {
-		require.NoError(t, db.Close())
-		require.NoError(t, os.RemoveAll(tempDir))
-	}
-}
-
 // TestRegisterSidecar makes sure that registering a sidecar ticket verifies the
 // offer signature contained within, adds the recipient node's information to
 // the ticket and stores it to the local database.
 func TestRegisterSidecar(t *testing.T) {
-	t.Parallel()
 
-	mockSigner := test.NewMockSigner()
-	mockWallet := test.NewMockWalletKit()
-	mockSigner.Signature = testOfferSig.Serialize()
-
-	acceptor := NewSidecarAcceptor(&SidecarAcceptorConfig{
-		SidecarDB:  nil,
-		AcctDB:     nil,
-		Signer:     mockSigner,
-		Wallet:     mockWallet,
-		NodePubKey: ourNodePubKey,
-		ClientCfg:  auctioneer.Config{},
-	})
+	emptySetter := func(
+		ticket *sidecar.Ticket, sc *test.MockSignerClient,
+		wc *test.MockWalletKitClient, store *sidecar.MockStore,
+	) {
+	}
 
-	existingTicket, err := sidecar.NewTicket(
-		sidecar.VersionDefault, 1_000_000, 200_000, 2016,
-		providerPubKey, false,
-	)
-	require.NoError(t, err)
+	emptyCheck := func(t *testing.T, ticket *sidecar.Ticket) {
+	}
 
 	testCases := []struct {
 		name        string
 		ticket      *sidecar.Ticket
 		expectedErr string
-		check       func(t *testing.T, ticket *sidecar.Ticket)
+		mockSetter  func(
+			ticket *sidecar.Ticket, sc *test.MockSignerClient,
+			wc *test.MockWalletKitClient, store *sidecar.MockStore,
+		)
+		check func(t *testing.T, ticket *sidecar.Ticket)
 	}{{
-		name: "ticket with ID exists",
+		name:        "unable to register sidecar if ticket does not have a valid state",
```

Comment:
A way to remove one level of indentation would be to define these test cases globally and not within the test function.

Reply:
Done, it is quite hard to keep  < 80 columns in table driven tests, I promise I will do my best :sweat_smile: